packages/sources/ens/src/endpoint/lookup.ts

```diff
@@ -0,0 +1,80 @@
+import { Requester, Validator, AdapterError } from '@chainlink/ea-bootstrap'
+import { Config, ExecuteWithConfig, InputParameters } from '@chainlink/types'
+import { ethers } from 'ethers'
+import { initializeENS } from '../utils'
+
+export const description = 'Look up information about a human friendly ENS domain name'
+
+export const supportedEndpoints = ['lookup']
+export const endpointResultPaths = {
+  lookup: 'address',
+}
+
+export const inputParameters: InputParameters = {
+  ensName: {
+    description: 'The ENS name to look up',
+    required: true,
+    type: 'string',
+  },
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+
+  const jobRunID = validator.validated.id
+  const name = validator.validated.data.ensName.toLowerCase()
+  const resultPath = validator.validated.data.resultPath
+
+  const splitName = name.split('.')
+  if (splitName.length < 2)
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid ENS name. Format must be [domain].[top level domain], (e.g. chainlink.eth)`,
+      statusCode: 400,
+    })
+
+  const networkProvider = new ethers.providers.JsonRpcProvider(config.rpcUrl)
+  const contracts = await initializeENS(networkProvider)
```

Comment:
Looking at the impl it seems to make a lot of async requests, would moving this initialization outside of the endpoint logic affect behavior?

Reply:
Yeah, I was thinking about that too. I'll initialize it on startup instead and hold the contracts in the config.